portal/unit_tests/domain_parsing.test.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect, test } from 'vitest'
+import { getDomain } from '../src/helpers/domain_parsing.ts'
+
+const testCases: [string, string][] = [
+    ['https://example.com', 'example.com'],
+    ['https://suinsname.localhost:8080', 'localhost'],
+    ['https://suinsname.subname.localhost:8080', 'localhost'],
```

Comment:
Important: Following standard DNS conventions, this should rather be `subname.suinsname.[...]`, also in the other subname examples.

Reply:
You mean it should be `['https://suinsname.subname.localhost:8080', 'suinsname.subname.localhost']`?

Reply:
No, I mean that we generally want the `subname` to be before the `suinsname`.

Reply:
When I am hitting this url http://flatland.localhost:8080/#0x49de81ca6b68b5956a26ad275e0a0e7fb3e7d0165134bdd713e4719c9be8522b I see on the console the following: `Portal domain and request domain:  localhost localhost`. 
So I presume we should only parse the domain and not the suins + subnames part of the url in this function.

Reply:
> So I presume we should only parse the domain and not the suins + subnames part of the url in this function.

Yes, I agree. But for any follow-up, we need to ensure that we parse the SuiNS name and any subnames correctly as well. And domain names start with the most specific name and then up to more general. So the `subname` needs to come before the `suinsname`. So, for example, `subsubname.subname.suinsname.somedomain.sometld`.

Reply:
Ok clear! I will update this, thanks @mlegner!